### PR TITLE
diagnostics: Fix panic when a NIC doesn't have a mac address

### PR DESCRIPTION
**Pull request by YKG**
Signed-off-by: Kaige Ye <yekaige2010@gmail.com>

### What problem does this PR solve?

Problem Summary:

`pnet_datalink::NetworkInterface::mac_address()` will panic if the NIC doesn't have a mac address. For example, a point-to-point link doesn't have mac address.

### What is changed and how it works?

What's Changed:

Return a string "none" when the `NIC.mac` is `None` 

### Check List

Tests

- Manual test

  * add a point-to-point interface (e.g. with `WireGuard` or `OpenVPN` or just `ip tunnel add`)
  * run `tiup playground`
  * visit http://localhost:2379/dashboard/#cluster-info/host
  * check tikv logs
      - without this patch, tikv panics
      - with this patch, no panic any more

### CODE REVIEW

File: src/server/service/diagnostics/sys.rs [hunk 1]
```diff
@@ -377,11 +377,18 @@ fn disk_hardware_info(collector: &mut Vec<ServerInfoItem>) {
...
+        None => "none".to_string(),
+    }
+}
+
 fn nic_hardware_info(collector: &mut Vec<ServerInfoItem>) {
     let nics = pnet_datalink::interfaces();
     for nic in nics.into_iter() {
         let mut infos = vec![
-            ("mac", nic.mac_address().to_string()),
+            ("mac", mac_address(nic.mac)),
```
Comment (BusyJay) [inline src/server/service/diagnostics/sys.rs:391]: When would these information useful?

Comment (YKG) [reply src/server/service/diagnostics/sys.rs:391]: I don't know who and when will the `mac` infomation will be used.
What I know is `nic_hardware_info` will be called by `hardware_info`, and `hardware_info` will be called when you visit PD's dashboard's `/#cluster-info/host`

Comment (BusyJay) [reply src/server/service/diagnostics/sys.rs:391]: /cc @breeswish can it be removed?

Comment (breeswish) [reply src/server/service/diagnostics/sys.rs:391]: They are printed in https://pingcap.com/docs-cn/stable/system-tables/system-table-cluster-hardware/ @BusyJay

**Comment by breeswish** [comment]
@lonng PTAL

**Comment by breeswish** [comment]
@YKG Hi, could you also propose a cherry pick PR over release-4.0 branch for your #7889  ?

**Comment by YKG** [comment]
@breeswish Sorry but I don't know how to it, is there any guidelines I can follow to do `propose a cherry pick PR over release-4.0 branch`?

**Comment by breeswish** [comment]
You can open a new PR that merge into release-4.0 instead of master, like this:

image

Notice that the new PR should based on release-4.0 branch as well to make changes.

**Pull request merged**